interfaces/builtin/allegro_vcu.go

```diff
@@ -0,0 +1,63 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://www.xilinx.com/support/documentation/ip_documentation/vcu/v1_2/pg252-vcu.pdf
+// https://github.com/Xilinx/vcu-modules/
+const allegroVcuSummary = `allows access to Xilinx Allegro Video Code Unit`
+
+// Xilinx offers IP for their devices to decode/encode video streams, by
+// using /dev/allegroDecodeIP and /dev/allegroIP devices.
+// These operations should be considered privileged since the driver
+// assumes trusted input, therefore require manual connection.
+const allegroVcuBaseDeclarationSlots = `
```

Comment:
hmm I wonder if this should be super-privileged, considering the low-level control that is provided to the kernel module. probably not, since the fpga interface is not super-privileged, but just wondering out loud.

Reply:
Note that there are a few interfaces that give access to devices that give that short of low-level control, being one of them the opengl one, which is even auto-connected...

Reply:
I was also wondering if this should be super privileged since the comment here says that is assumes trusted input, however the existing vcio interface also has this same stipulation and is only manually connected, not super-privileged so there is precedent for this. Note the opengl interface does not assume the input is trusted to my knowledge - so this is a different level of trust between the two.

Reply:
TBH I think I copied the comment from the vcio interface because because that interface provides similar functionality for the RPi. So maybe the comment is not fully accurate and the zocl driver is safer than this, but I really don't know.